index.d.ts

```diff
@@ -0,0 +1,170 @@
+declare module 'fluture' {
+  interface ResolveFunction<R> {
+    (value: R): void
+  }
+  interface RejectFunction<L> {
+    (error: L): void
+  }
+
+  interface Cancel {
+    (): void
+  }
+
+  export abstract class IFuture<L, R> {
+    fork(reject: RejectFunction<L>, resolve: ResolveFunction<R>): Cancel
+  }
+
+  export function after<L, R>(duration: number, value: R): IFuture<L, R>
+
+  export function and<L, R>(
+    left: IFuture<L, any>,
+    right: IFuture<L, R>
+  ): IFuture<L, R>
+
+  export function ap<L, R, VR>(
+    apply: IFuture<L, (value: VR) => R>,
```

Comment:
My suggestion would be to specialise `Future.ap` too, since using a particular encoding for HKTs would couple this library with that implementation. 

Writing the bindings of `Fluture` for `fp-ts` is actually pretty easy and could be done in userland

```ts
//
// fp-ts bindings for Fluture
//

import { ap, of, map, chain, IFuture } from 'fluture'
import { Monad } from 'fp-ts/lib/Monad'

declare module 'fp-ts/lib/HKT' {
  interface URI2HKT2<L, A> {
    Fluture: IFuture<L, A>
  }
}

declare module 'fluture' {
  interface IFuture<L, R> {
    _A: R
    _L: L
    _URI: URI
  }
}

export const URI = 'Fluture'

export type URI = typeof URI

export const monad: Monad<URI> = {
  URI,
  map,
  of,
  ap,
  chain
}

//
// usage
//

import { lift } from 'fp-ts/lib/Functor'

const double = (n: number) => n * 2

// liftedDouble: <L>(fa: IFuture<L, number>) => IFuture<L, number>
export const liftedDouble = lift(monad, double)
```